contracts/contracts/compile.go

```diff
@@ -4,13 +4,21 @@
 package contracts
 
 // Step 1: Compile Solidity contracts to generate ABI and bin files
-//go:generate solc-v0.8.30 -o ../artifacts --overwrite --abi --bin --base-path . @openzeppelin/contracts/=../node_modules/@openzeppelin/contracts/ AllowList.sol ERC20NativeMinter.sol ExampleDeployerList.sol ExampleFeeManager.sol ExampleRewardManager.sol ExampleTxAllowList.sol ExampleWarp.sol
+//go:generate solc-v0.8.30 -o ../artifacts --overwrite --abi --bin --base-path . --evm-version paris @openzeppelin/contracts/=../node_modules/@openzeppelin/contracts/ AllowList.sol ERC20NativeMinter.sol ExampleDeployerList.sol ExampleFeeManager.sol ExampleRewardManager.sol ExampleTxAllowList.sol ExampleWarp.sol interfaces/IAllowList.sol interfaces/IFeeManager.sol interfaces/INativeMinter.sol interfaces/IRewardManager.sol interfaces/IWarpMessenger.sol
```

Comment:
We are compiling to `paris`, rather than `shanghai` because the `PUSH0` opcode introduced in Shanghai causes "invalid opcode" errors. This seems fine for now, but as the tests are about behavior, not gas optimization, but this should be investigated further. I think that perhaps libevm's backend stuff is out of date? 

This is why all the tests are slightly larger in bytecode is because we do not have this optimization anymore.

Reply:
Sounds good! Once that gets merged, I'll change the target to Shanghai which will match the current tests.